syntax = "proto3";

package aea.fetchai.consensus.v0_1_0;

message ConsensusMessage{

  // Performatives and contents
  message Observation_Performative{
    int32 value = 1;
    int32 time = 2;
    string source = 3;
    string signature = 4;
  }

  message Aggregation_Performative{
    int32 value = 1;
    int32 time = 2;
    repeated int32 contributors = 3;
    string signature = 4;
  }


  oneof performative{
    Aggregation_Performative aggregation = 5;
    Observation_Performative observation = 6;
  }
}
